### ISSUE TITLE
Several questions about the paper and code

### ISSUE THREAD

**Issue by JasonCZH4** [Jin-Ying/Multi-Level-Logit-Distillation#2]
I like you work. It 's very clear and intuitive.
But have few questions:
1. Why did you use data augmentation which were not included in DKD?  Should data augmentation ablation experiments be included?
2. Why is the training epoch set to 480, while DKD's original experiment was only 240?
3. Why not compare with DKD in the Cifar100 experiment?
4. Why is data augmentation a random selection rather than fixed?
5. Why use both weak data and strong data to calculate the Loss function?

Hope can get a reply! Thanks!

**Comment by JasonCZH4** [comment]
I find another problem.
```
self.ce_loss_weight = cfg.KD.LOSS.CE_WEIGHT
self.kd_loss_weight = cfg.KD.LOSS.KD_WEIGHT
```
I can not find the above hyperparameters in the config file.

**Comment by Swiminn** [comment]
> I find another problem.
> 
> ```
> self.ce_loss_weight = cfg.KD.LOSS.CE_WEIGHT
> self.kd_loss_weight = cfg.KD.LOSS.KD_WEIGHT
> ```
> 
> I can not find the above hyperparameters in the config file.


You can find it from mdistiller/engine/cfg.py
```
CFG.KD.LOSS.CE_WEIGHT = 0.1
CFG.KD.LOSS.KD_WEIGHT = 0.9
```

**Comment by JasonCZH4** [comment]
Thanks! I have found it just after I post the issue. However, I am still confused about the above 5 questions. lol

**Comment by Swiminn** [comment]
I agree with your first 5 questions.

I think adding data augmentation, or increasing training epochs needs reasonable justification

I'm also waiting for the author's answer to your question.

**Comment by JinYu1998** [comment]
> I find another problem.我发现了另一个问题。
> 
> ```
> self.ce_loss_weight = cfg.KD.LOSS.CE_WEIGHT
> self.kd_loss_weight = cfg.KD.LOSS.KD_WEIGHT
> ```
> 
> I can not find the above hyperparameters in the config file.我在配置文件中找不到上述超参数。

I think you may need to refer to the DKD code  : https://github.com/megvii-research/mdistiller

**Comment by JinYu1998** [comment]
> I like you work. It 's very clear and intuitive.我喜欢你的工作。它非常清晰直观。 But have few questions:但有几个问题：
> 
> 1. Why did you use data augmentation which were not included in DKD?  Should data augmentation ablation experiments be included?为什么使用DKD中未包含的数据增强？是否应该包括数据增强消融实验？
> 2. Why is the training epoch set to 480, while DKD's original experiment was only 240?为什么训练周期设置为 480，而 DKD 的原始实验只有 240？
> 3. Why not compare with DKD in the Cifar100 experiment?为什么不与Cifar100实验中的DKD进行比较呢？
> 4. Why is data augmentation a random selection rather than fixed?为什么数据增强是随机选择而不是固定的？
> 5. Why use both weak data and strong data to calculate the Loss function?为什么要同时使用弱数据和强数据来计算损失函数？
> 
> Hope can get a reply! Thanks!希望能得到答复！谢谢！

Same question, why are learning rates and Epochs not consistent with other KD methods.

**Comment by helloworld77** [comment]
I remove the strong augmentation, It could not achieve a better acc than dkd. I think augmentation is key to the high score.

**Comment by zzeoZheng** [comment]
> I remove the strong augmentation, It could not achieve a better acc than dkd. I think augmentation is key to the high score.

yeah, I removed the data strong augmentation too, and  it could not achieve a better acc than the result of the paper ,and even much less than it.

**Comment by zhiweihu1103** [comment]
I found the same problem. I think the author should give a reasonable explanation.

**Comment by JasonCZH4** [comment]
After so many people and such a long time of verification, I believe that this paper is completely fake and has no academic value. If possible, it should be withdrawn from CVPR2023. 👎

**Comment by Jin-Ying** [comment]
Sorry for not noticing this question. I will remove the data augmentation in the code. You need to tune the temperature value and loss trade-off to obtain the results in the paper.

**Comment by Jin-Ying** [comment]
If you have any more questions, please contact me via my email. I am happy to have discussions with you.

**Comment by JasonCZH4** [comment]
> Sorry for not noticing this question. I will remove the data augmentation in the code. You need to tune the temperature value and loss trade-off to obtain the results in the paper.

Obviously, tuning the temperature is a trick to get performance improvement. I don't think this is a  thorough solution to the five problems. Maybe you should tune the temperature and loss yourself, and tell us what REAL result is. The current results are unreasonable at all.

**Comment by Jin-Ying** [comment]
Thank you for raising the questions above. I sincerely apologize again for my late response to these questions. 

First of all, I would like to clarify that all the results reported in our CVPR paper can be reproduced using the released code. That being said, there might be implementation details which we did not explain clearly in the paper. I am more than willing to provide more information and results to assist the reproduction or verification of our results. 

Please allow me to first address the questions briefly here. Additional details and results will be posted as an organized document in a few days.

For questions 1, 4 and 5 (about the choice of data augmentation strategy): different data augmentation strategies may work optimally for different knowledge distillation methods. In the paper, we presented the strategy that aligns the best with the proposed method. We will provide a comprehensive set of ablation results. 

For question 2 (about training epochs): the number of training epoches is a hyperparameter in knowledge distillation, whose optimal setting may vary for different methods. We will provide more ablation results to show how this setting affects the performance. 

For question 3 (about the comparison with DKD on CIFAR-100): please note that the paper has presented the comparative results on ImageNet, which, we believed, has clear demonstrated the benefit of our proposed method, as ImageNet is more diverse and challenging. That being said, we will also provide the results on CIFAR-100, in order to clarify the concern above. 

Sorry for the confusion caused due to the omission of certain ablation results in the paper. As said above, we will post a complementary report in a more organized manner in a few days.

**Comment by JasonCZH4** [comment]
Thanks for your reply! Looking forward to your report.

**Comment by JinYu1998** [comment]
I look forward to seeing the results of experiments that strictly refer to the DKD hyperparameter settings in your report. 👍

**Comment by shaoshitong** [comment]
I think the author is deflecting attention to cover up the fakery, and while it's hard to keep going deeper into traditional KD to this day, I hope the author admits it in a big way, or at least offers some contribution to the KD community.

**Comment by zzeoZheng** [comment]
> I think the author is deflecting attention to cover up the fakery, and while it's hard to keep going deeper into traditional KD to this day, I hope the author admits it in a big way, or at least offers some contribution to the KD community.

I agree with your point of view. The author's reply deliberately avoids the key point of the problem and covers up the fact of fraud! In fact, the reason why the results can be reproduced is because of data augmentation. After removing it, the results are even worse than any work five years ago. Secondly, the author guides us to focus on her remaining ablation experiments in her reply, but this is not the key to our problem. Our key is whether the method of this article is effective under the same experimental setting, just like the results in the paper. If the method itself is effective and not some kind of cheating, why did the author deliberately avoid it in the paper? We need a reasonable explanation and look forward to the author's follow-up reply.

**Comment by Jin-Ying** [comment]
Thank you for your insightful suggestions. Here I provide the following preliminary results, and I hope that they can address your concerns.

Table 1: Results on Cifar-100, Homogeneous Architecture. Average Top-1 accuracy is reported. (The same as Table 2 in our paper)
| Method | Epoch | Aug |  Avg     |
| ------ | ----- | --- | ----- |
| DKD    | 240   | No  | 74.69 |
| MLLD   | 240   | No  | 74.72 |
| DKD    | 240   | Yes | 74.26 |
| MLLD   | 240   | Yes | 75.12 |
| DKD    | 480   | Yes | 74.35 |
| MLLD   | 480   | Yes | 75.09 |

Table2: Results on ImageNet, ResNet34->ResNet18 and ResNet50->MobileNet-V2. Top-1 and Top-5 accuracy is reported. (The same as Table 4 in our paper)
| Method | Epoch | Aug | Top-1 | Top-5 | Top-1 | Top-5 |
| ------ | ----- | --- | ----- | ----- | ----- | ----- |
| DKD    | 100   | No  | 71.7  | 90.41 | 72.05 | 91.05 |
| MLLD   | 100   | Yes | 71.87 | 90.5  | 72.89 | 91.45 |
| DKD    | 100   | Yes | 71.74 | 90.43 | 71.85 | 90.80 |
| MLLD   | 100   | Yes | 71.90 | 90.55 | 73.01 | 91.42 |

**1. Performance of our method with identical setting of DKD**
We perform fair comparsions between DKD and our method (MLLD) in various settings.
* In Table 1 Line 1~Line 2: Both MLLD and DKD have 240 training epoch and no strong data augmentation on Cifar-100. MLLD shows slightly better performance than DKD.

* In Table 1 Line 3~Line 4: Both MLLD and DKD have 240 training epoch and the same strong data augmentation on Cifar-100. MLLD shows clearly better performance than DKD.

* In Table 1 Line 5~Line 6: Both MLLD and DKD have 480 training epoch and the same strong data augmentation on Cifar-100. MLLD shows clearly better performance than DKD.

* In Table 2 Line 1~Line 2: Both MLLD and DKD have 100 training epoch and no strong data augmentation on ImageNet. MLLD shows clearly better performance than DKD.

* In Table 2 Line 3~Line 4: Both MLLD and DKD have 100 training epoch and the same strong data augmentation on ImageNet. MLLD shows clearly better performance than DKD.

We can summarize the observations as follows:
1. On Cifar-100, MLLD is slightly better than DKD without augmentation. When both MLLD and DKD have strong augmentation, MLLD is clearly better. On ImageNet, MLLD is consistently better than DKD.
2. MLLD has a good property to work together with strong augmentation for better performance. However, DKD can not benefit from strong augmentation even if we have tuned the hyperparameters for it.

Notably, the learning rate and temperature of DKD with strong augmentation and MLLD without strong augmentation are further tuned to be suitable with new settings, respectively.

There may exist arguments whether the learning rate and temperature should be tuned in different settings. We believe that tuning hyperparameters is necessary to sufficiently show the potential of the method. Moreover, we also tune the hyperparameters in DKD with augmentation and report the best performance.

**2. Epoch number**
First, we shall mention that on ImageNet, our training epoch is identical with previous methods (including DKD). The difference lies in Cifar100.

We have also provided the ablation study about training epochs (240 and 480, respectively) on MLLD, as in Table 1 Line 4 & 6.

We can observe that the longer training epochs only lead to slightly performance variance. Our method shows consistent competitive performance.

We will update MLLD performance with 240 epoch on Cifar100 in our arxiv version.

**3. Discussion about the strong augmentation**
We apply the strong data augmentation on DKD and discover that it sometimes has negative influences on the performance (74.69 -> 74.26) on Cifar100. We conjecture that augmented data brings about more uncertainty on model predictions, and our method is likely to be more robust to such uncertainty. 

In addition, We notice that data augmentation brings about more performance gains for MLLD on Cifar100 than ImageNet. We conjecture that for small-scale datasets, data augmentation will alleviate overfitting and enhance model robustness.

**4. Our promise**
Finally, sincere gratittude to all of you again for your suggestions. We will update the released code, configs and arxiv paper, with detailed ablation studies and implementation details for both MLLD and DKD. Thank you for your patience!
